test/built-ins/Array/prototype/concat/Array.prototype.concat_array-like-string-length.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2014 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flags: --harmony-arrays --harmony-classes
+
+/*---
+es6id: 22.1.3.1_3
+description: Array.prototype.concat array like string length
+includes: [compareArray.js]
+---*/
+var obj = {
+  "length": "6",
+  "1": "A",
+  "3": "B",
+  "5": "C"
+};
+obj[Symbol.isConcatSpreadable] = true;
+var obj2 = { length: 3, "0": "0", "1": "1", "2": "2" };
+var arr = ["X", "Y", "Z"];
+
+var expected = [
+  void 0, "A", void 0, "B", void 0, "C",
+  { "length": 3, "0": "0", "1": "1", "2": "2" },
```

Comment:
there's no reason it couldn't be replaced with obj2, tbh --- it should be a reference to the same object. Don't ask me why I didn't write it that way o_O